BankAccount/account.js

```diff
@@ -0,0 +1,107 @@
+class AccountBase {
+    #account_name
+    #bank_name
+    #balance
+    #created
+    #updated
+
+    constructor(account_name, bank_name, balance = 0) {
+        if (new.target === AccountBase) {
+            throw new Error("Cannot instantiate abstract class Animal directly")
+        }
+        this.#account_name = account_name
+        this.#bank_name = bank_name
+        this.#balance = balance
+        this.#created = new Date()
+        this.#updated = null
+    }
+
+    withdraw(amount) {
+        if (this.#balance >= amount) {
+            this.#balance -= amount;
+            this.#created = new Date()
+            console.log(console.log(`Detail: ₦${amount} succesfully withdrawn.`))
+        } else {
+            console.log("Withdrawal failed")
+        }
+    }
+
+    deposit() {
+        throw new Error("Must be implemented by a subclass");
+    }
+
+    balance() {
+        return this.#balance
+    }
+
+    info() {
+        console.log(`Bank name: ${this.#bank_name}`)
+        console.log(`Account name: ${this.#account_name}`)
+        console.log(`Account balance: ₦ ${this.#balance}`)
+        console.log(`Account created at: ${this.#created}`)
+        console.log(`Account updated at: ${this.#updated}`)
+    }
+}
+
+class AdultAccount extends AccountBase {
+
+    deposit(amount) {
+        this.balance += amount
+        this.created = new Date()
+        console.log("Deposit succesful")
+    }
+}
+
+class StudentAccount extends AccountBase {
+
+    static ACCOUNT_LIMIT = 4E5;
+
+
+    deposit(amount) {
+        if ((this.balance + amount > this.ACCOUNT_LIMIT)) {
+            console.log("Account limit exceeded.")
+        } else {
+            this.balance += amount
+            this.created = new Date()
+            console.log("Deposit succesful")
+        }
+
+    }
+}
+
+class Bank {
+    #name
+    #accounts
+
+    constructor(name) {
+        this.#name = name
+        this.#accounts = []
+    }
+
+    create_adult_account(account_name, initial_deposit) {
+        const acct = new AdultAccount(account_name, this.#name, initial_deposit)
+        this.#accounts.push(acct)
+        return acct
+    }
+
+    list_accounts() {
+        return this.#accounts
+    }
+
+    total_amount() {
+        const total = this.#accounts.reduce((sum, acct) => sum + acct.balance(), 0);
```

Comment:
When I run the program it run successfully but I get this error in the end:

```code
/home/eteims/bank.js:92
        const total = this.#accounts.reduce((sum, acct) => sum + acct.balance(), 0);
                                                                      ^

TypeError: acct.balance is not a function
    at /home/eteims/bank.js:92:71
    at Array.reduce (<anonymous>)
    at Bank.total_amount (/home/eteims/bank.js:92:38)
    at Object.<anonymous> (/home/eteims/bank.js:106:42)
    at Module._compile (node:internal/modules/cjs/loader:1101:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1153:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
    at node:internal/main/run_main_module:17:47
```

There seems to be a conflict of names because when I change the blance function name. it works.